applications/StructuralMechanicsApplication/custom_strategies/custom_schemes/explicit_central_differences_scheme.hpp

```diff
@@ -0,0 +1,900 @@
+          array_1d<double,3>& r_current_velocity        = i->FastGetSolutionStepValue(VELOCITY);
+          array_1d<double,3>& r_current_displacement    = i->FastGetSolutionStepValue(DISPLACEMENT);
+          array_1d<double,3>& r_middle_velocity         = i->GetValue(MIDDLE_VELOCITY);
+
+          array_1d<double,3>& r_current_acceleration    = i->FastGetSolutionStepValue(ACCELERATION);
+
+          //Solution of the explicit equation:
+          if (nodal_mass > numerical_limit)  r_current_acceleration = r_current_residual/nodal_mass;
+          else r_current_acceleration = ZeroVector(3);
+
+
+
+          int DoF = 2;
+          bool fix_displacements[3] = {false, false, false};
+
+          fix_displacements[0] = (i->pGetDof(DISPLACEMENT_X))->IsFixed();
+          fix_displacements[1] = (i->pGetDof(DISPLACEMENT_Y))->IsFixed();
+
+          if( i->HasDofFor(DISPLACEMENT_Z) )
+          {
+            DoF = 3;
+            fix_displacements[2] = (i->pGetDof(DISPLACEMENT_Z))->IsFixed();
+          }
+
+          for (int j = 0; j < DoF; j++) 
+          {
+              
+              if (fix_displacements[j] == true) 
+              {
+                  
+                r_current_acceleration[j]  = 0.0;
+                r_middle_velocity[j]       = 0.0; 
+                
+              }
+              
+              r_current_velocity[j]      = r_middle_velocity[j] + (mTime.Previous - mTime.PreviousMiddle) * r_current_acceleration[j]; //+ actual_velocity;
+              r_middle_velocity[j]       = r_current_velocity[j] + (mTime.Middle - mTime.Previous) * r_current_acceleration[j] ; 
+              r_current_displacement[j]  = r_current_displacement[j] + mTime.Delta * r_middle_velocity[j];      
+              
+              
+          }//for DoF
+
+
+          ////// ROTATION DEGRESS OF FREEDOM
+          if (i->HasDofFor(ROTATION_X))
+          {
+            array_1d<double,3> nodal_inertia     = i->GetValue(NODAL_INERTIA);  
+            array_1d<double,3>& r_current_residual_moment          = i->FastGetSolutionStepValue(MOMENT_RESIDUAL);
+            array_1d<double,3>& r_current_angular_velocity         = i->FastGetSolutionStepValue(ANGULAR_VELOCITY);
+            array_1d<double,3>& r_current_rotation                 = i->FastGetSolutionStepValue(ROTATION);
+            array_1d<double,3>& r_middle_angular_velocity          = i->GetValue(MIDDLE_ANGULAR_VELOCITY);
+            array_1d<double,3>& r_current_angular_acceleration     = i->FastGetSolutionStepValue(ANGULAR_ACCELERATION);
+
+            for (int kk = 0; kk<3; ++kk)
+            {         
+              if (nodal_inertia[kk] > numerical_limit)  r_current_angular_acceleration[kk] = r_current_residual_moment[kk] / nodal_inertia[kk];
+              else r_current_angular_acceleration[kk] = 0.00;
+            }
+            
+
+            DoF = 2;
+            bool fix_rotation[3] = {false, false, false};
+            fix_rotation[0] = (i->pGetDof(ROTATION_X))->IsFixed();
```

Comment:
i would use the version which knows the key (As it is done in the fluid elements).
It is MUCH faster